juju/juju
# [JUJU-2334] Add state logic and persistence to support storing and watching secret backend rotations

Add support in the state layer to support managing and watching secret backend rotations.
A new collection is used to store the next rotate time for the backends, and a watcher is provided to notify of changes.
The watcher implementation is based on what was done for secret rotations.

Rotation support is added to the vault provider (the only one that supports it).

The next PR will wire up the worker and api facade to actually do the rotations.

## Checklist


## QA steps

Add a vault backend to a controller.
Update the backend to use token rotation, eg juju update-secret-backend myvault token-rotate=2h
Remove the token rotation, eg juju update-secret-backend myvault token-rotate=0s
Nothing triggers on this yet. I checked the mongo collection to see the results.
